src/ProjectTemplates/Web.ProjectTemplates/content/ComponentsWebAssembly-CSharp/Client/Pages/ShowProfile.razor

```diff
@@ -0,0 +1,36 @@
+﻿@page "/showprofile"
+@using Microsoft.AspNetCore.Authorization
+@using Microsoft.AspNetCore.Components.WebAssembly.Authentication
+@using ComponentsWebAssembly_CSharp.Shared
+@attribute [Authorize]
+@inject HttpClient Http
+
+<h1>Call Graph</h1>
+
+<p>This component demonstrates fetching data from the server.</p>
+
+@if (profile == null)
+{
+    <p><em>Loading...</em></p>
+}
+else
+{
+    @profile
+}
+
+@code {
+    private string profile;
+
+    protected override async Task OnInitializedAsync()
+    {
+        try
+        {
+            profile = await Http.GetStringAsync("showProfile");
```

Comment:
Why would the webassembly version go through the server to call graph instead of doing it directly from the client?
Right now is Browser -> Server -> Graph while it is most common to do Browser -> Graph and not involve the server at all.

That would work for non hosted apps too.

Reply:
Because the Graph API is available through the Graph SDK on the server, not on the SPA.
Having said that, yes, you should probably provide an implementation for the non-hosted case.

Reply:
@jmprieur Is there ar reason why we couldn't use the graph SDK directly from the client app in WebAssembly?

Reply:
@javiercn : maybe we want to sync-up next week (in our time zone) to see if we can do something quick here?

Reply:
That sounds good